test/general/lambda_naming.pass.cpp

```diff
@@ -58,7 +60,13 @@ int main() {
     EXPECT_TRUE(!is_equal, "wrong return value from equal");
     auto does_1_exist = ::std::find(policy, buf_begin, buf_end, 1);
     EXPECT_TRUE(does_1_exist - buf_begin == 1000, "wrong return value from find");
-#endif
+#endif // !_ONEDPL_FPGA_DEVICE
+
+#elif !__SYCL_UNNAMED_LAMBDA__ && _ONEDPL_BACKEND_SYCL
```

Comment:
Since we have test cases for both `__SYCL_UNNAMED_LAMBDA__` and `!__SYCL_UNNAMED_LAMBDA__`, let's move `_ONEDPL_BACKEND_SYCL` above, e.g.:
```
#if _ONEDPL_BACKEND_SYCL
#if __SYCL_UNNAMED_LAMBDA__ 
...
#endif
#if !__SYCL_UNNAMED_LAMBDA__ 
...
#endif
#endif // _ONEDPL_BACKEND_SYCL
```